gson/src/test/java/com/google/gson/functional/Java17RecordTest.java

```diff
@@ -43,28 +42,42 @@ public void testMultipleNamesDeserializedCorrectly() {
     assertEquals("v1", gson.fromJson("{'name':'v1'}", MyRecord.class).a);
 
     // Both name1 and name2 gets deserialized to b
-    assertEquals("v11", gson.fromJson("{'name1':'v11'}", MyRecord.class).b);
-    assertEquals("v2", gson.fromJson("{'name2':'v2'}", MyRecord.class).b);
-    assertEquals("v3", gson.fromJson("{'name3':'v3'}", MyRecord.class).b);
+    assertEquals("v11", gson.fromJson("{'name': 'v1', 'name1':'v11'}", MyRecord.class).b);
+    assertEquals("v2", gson.fromJson("{'name': 'v1', 'name2':'v2'}", MyRecord.class).b);
+    assertEquals("v3", gson.fromJson("{'name': 'v1', 'name3':'v3'}", MyRecord.class).b);
   }
 
   @Test
   public void testMultipleNamesInTheSameString() {
     // The last value takes precedence
-    assertEquals("v3", gson.fromJson("{'name1':'v1','name2':'v2','name3':'v3'}", MyRecord.class).b);
+    assertEquals("v3",
+        gson.fromJson("{'name': 'foo', 'name1':'v1','name2':'v2','name3':'v3'}", MyRecord.class).b);
   }
 
   @Test
   public void testConstructorRuns() {
-    assertThrows(NullPointerException.class,
-        () -> gson.fromJson("{'name1': null, 'name2': null", MyRecord.class));
+    Throwable e = assertThrows(RuntimeException.class,
+        () -> gson.fromJson("{'name1': null, 'name2': null}", MyRecord.class));
+    while (e != null && !(e instanceof NullPointerException)) {
+      e = e.getCause();
+    }
+    assertNotNull(e);
   }
 
-  private static record MyRecord(
+  @Test
+  public void testPrimitiveDefaultValues() {
+    RecordWithPrimitives zero = new RecordWithPrimitives((byte) 0, (short) 0, 0, 0, 0, 0, '\0', false);
+    assertEquals(zero, gson.fromJson("{}", RecordWithPrimitives.class));
```

Comment:
Maybe for completeness one test with explicit JSON `null` for each component would be useful as well? (combined with the comment above, it could then for example use `{"s": "nulls", "aByte": null, ...}`)

Reply:
Done, though I'm kind of unhappy that an explicit `null` in the JSON produces these default values. I think that should probably produce a `JsonParseException`.

Reply:
That is a good point. It would probably be safe to make it strict initially and in case users complain we could still consider restoring the behavior with default values for JSON null.

Should I make that change in a separate PR?

Reply:
I would be very happy to see that!